sdktests/server_side_events_experimentation.go

```diff
@@ -34,26 +33,26 @@ func doServerSideExperimentationEventTests(t *ldtest.T) {
 			},
 		},
 	}
-	user := ldcontext.New("user-key")
+	context := ldcontext.New("user-key")
 
 	scenarios := []struct {
 		name           string
 		flagConfig     func(*ldbuilders.FlagBuilder)
-		expectedReason ldreason.EvaluationReason
+		expectedReason string
 	}{
 		{
 			name: "experiment in rule",
 			flagConfig: func(f *ldbuilders.FlagBuilder) {
 				f.AddRule(ldbuilders.NewRuleBuilder().ID("rule1").VariationOrRollout(rollout))
 			},
-			expectedReason: ldreason.NewEvalReasonRuleMatchExperiment(0, "rule1", true),
+			expectedReason: `{"kind": "RULE_MATCH", "ruleIndex": 0, "ruleId": "rule1", "inExperiment": true}`,
 		},
```

Comment:
question: why did this need to be changed?

Reply:
The effect is the same, but I felt like it would be better to spell out the expectation rather than relying on the Go SDK types and the fact that serializing them produces the expected JSON. The cross-platform specification is written in terms of the JSON schema.